vars/neoDeploy.groovy

```diff
@@ -8,222 +11,218 @@ import com.sap.piper.tools.ToolDescriptor
+    // Backward compatibility: ensure old configuration is taken into account
+    // The old configuration in not stage / step specific
+
+    def defaultDeployHost = script.commonPipelineEnvironment.getConfigProperty('DEPLOY_HOST')
+    if (defaultDeployHost) {
+        echo "[WARNING][${STEP_NAME}] A deprecated configuration framework is used for configuring parameter 'DEPLOY_HOST'. This configuration framework will be removed in future versions."
+        neoCompatibilityConfiguration.put('host', defaultDeployHost)
+    }
+
+    def defaultDeployAccount = script.commonPipelineEnvironment.getConfigProperty('CI_DEPLOY_ACCOUNT')
+    if (defaultDeployAccount) {
+        echo "[WARNING][${STEP_NAME}] A deprecated configuration framework is used for configuring parameter 'DEPLOY_ACCOUNT'. This configuration framekwork will be removed in future versions."
+        neoCompatibilityConfiguration.put('account', defaultDeployAccount)
+    }
+
+    if (parameters.deployHost && !parameters.host) {
+        echo "[WARNING][${STEP_NAME}] Deprecated parameter 'deployHost' is used. This will not work anymore in future versions. Use parameter 'host' instead."
+        parameters.put('host', parameters.deployHost)
+    }
+
+    if (parameters.deployAccount && !parameters.account) {
+        echo "[WARNING][${STEP_NAME}] Deprecated parameter 'deployAccount' is used. This will not work anymore in future versions. Use parameter 'account' instead."
+        parameters.put('account', parameters.deployAccount)
+    }
+
+    def credId = script.commonPipelineEnvironment.getConfigProperty('neoCredentialsId')
+    if (credId && !parameters.neoCredentialsId) {
+        echo "[WARNING][${STEP_NAME}] Deprecated parameter 'neoCredentialsId' from old configuration framework is used. This will not work anymore in future versions."
+        parameters.put('neoCredentialsId', credId)
+    }
+
+    if (!neoCompatibilityConfiguration.isEmpty()) {
+        echo "[WARNING][$STEP_NAME] You are using a deprecated configuration framework. This will be removed in " +
+            'futureVersions.\nAdd snippet below to \'./pipeline/config.yml\' and remove ' +
+            'file \'.pipeline/configuration.properties\'.\n' +
+            """|steps:
+                    |    neoDeploy:
+                    |        neo:
+                    |            host: ${neoCompatibilityConfiguration.get('host', '<Add host here>')}
+                    |            account: ${neoCompatibilityConfiguration.get('account', '<Add account here>')}
+                """.stripMargin()
+
+        if (Boolean.getBoolean('com.sap.piper.featureFlag.buildUnstableWhenOldConfigFrameworkIsUsedByNeoDeploy')) {
+            script.currentBuild.setResult('UNSTABLE')
+            echo "[WARNING][$STEP_NAME] Build has been set to unstable since old config framework is used."
         }
+        return [neo: neoCompatibilityConfiguration]
+    }
+
+    return [:]
+}
+
+private assertPasswordRules(String password) {
```

Comment:
Such kind of password checks are rather uncommon. What is the story behind that? Was there any bad experience with invalid passwords?

Reply:
Yes. The issue with the neo tool is, that in case you have a password starting with @ you will get a file not found exception. Name of the file is the rest of the password which is also shown in the log as plain text, ...